### ISSUE TITLE
Missing d.ts files

### ISSUE THREAD

**Issue by TeraNovaLP** [developit/microbundle#669]
Hello, I've encountered a problem with the Typescript definition files.

In the affected project definitions are created for all files except two, which I can't explain.

Thanks for your help!

Affected project: https://gitlab.com/sokkuri/kogitte

I miss the definitions for these files:
- models/AuthData.ts
- models/SignInData.ts

**Comment by marvinhagemeister** [comment]
Can confirm that this is an upstream bug in https://github.com/ezolenko/rollup-plugin-typescript2 and not in `microbundle` . Also tried Using TypeScript directly without a bundler works fine. My guess is that the generation emit happens too late or that the file is treated as being empty (the class is only used as an interface in the code).

Moving the unused classes into `AuthResult.ts` seems to work around that problem, but that's a workaround at best. The issue should really be fixed over there.

**Comment by saurabhnemade** [comment]
I have also faced this problem. 
What I ended up doing was introducing new script in package.json and run it manually when I need to emit typess:

```
"copy-dts": "copyfiles -u 1 \"src/**/*.d.ts\" dist",
```

But I would also like a permanent solution with which I can just run my own command so I can copy types or package should do this.

**Comment by hbroer** [comment]
Same problem here. I have a types.ts file which only exports a few types so the implementation file is not full of types and this file is not exported to types.d.ts. I will move the types to the implementation until it got fixed.

**Comment by geemanjs** [comment]
I hit the same problem.. For anyone just arriving here:

Seems to affect `12.0.1` on and above. 

I downgraded `microbundle` to the last known working version 👍 
```
npm install --save microbundle@12.0.0
```

**Comment by AndyOGo** [comment]
downgrading to `npm install --save-dev microbundle@0.12.0`

**Comment by developit** [comment]
Microbundle 0.13 includes a new version of `rollup-plugin-typescript2`, hopefully it fixes this bug.

**Comment by rhyek** [comment]
I'm using 0.13 and a file with only types exports is not being emitted. Tried 0.12 and it does emit them now.

**Comment by rschristian** [comment]
@rhyek Can you open a new issue? The problem outlined here was corrected. Whatever you're seeing is not the same thing, and we'd need information on your setup / what your issue is in order to provide help.

**Comment by agilgur5** [comment]
> I'm using 0.13 and a file with only types exports is not being emitted. Tried 0.12 and it does emit them now.

Thought I'd update here that I fixed the root cause upstream in https://github.com/ezolenko/rollup-plugin-typescript2/pull/406, which was released in rpt2 [`0.34.0`](https://github.com/ezolenko/rollup-plugin-typescript2/releases/tag/0.34.0).

There shouldn't be a difference between those two `microbundle` versions though, as this has been a long-standing issue in rpt2 since day 1 basically. There may be a difference with your `tsconfig` `include` though.
